cpp/src/sort/sort.cu

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/sorting.hpp>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/table/table_view.hpp>
+#include <utilities/error_utils.hpp>
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <thrust/sequence.h>
+
+namespace cudf {
+namespace exp {
+
+namespace detail {
+
+// Create permuted row indices that would materialize sorted order
+std::unique_ptr<column> sorted_order(table_view input,
+                                     std::vector<order> const& column_order,
+                                     null_order null_precedence,
+                                     cudaStream_t stream = 0) {
+  if (input.num_rows() == 0 or input.num_columns() == 0) {
+    return cudf::make_numeric_column(data_type{INT32}, 0);
```

Comment:
How will this work with non-numeric columns? How to call the right factory?

Reply:
The output of this function is a list of indices of the sorted order of the table. Therefore, it will always be a numeric column.